tests/test_username_api.py

```diff
@@ -13,23 +12,33 @@
 data = yaml.load(open(os.path.join('tests', 'test_data.yml')))
 websites = yaml.load(open('websites.yml'))
 
-invalid_username = '$very%long{invalid}user(name)'
+AvailabilityTestCase = collections.namedtuple(
+    'AvailabilityTestCase',
+    'website username')
 
 
-def generate_random_username(website, length):
-    first_char_letter = False
-    if website == 'openhub':
-        first_char_letter = True
-    return ''.join(generate_random_character(length,
-                                             first_char_letter))
+@pytest.fixture(params=data.keys())
+def random_available_min_username(request):
+    return AvailabilityTestCase(
+        request.param,
+        generate_random_valid_username(
+            request.param,
+            websites['username_patterns'][request.param]['available_min_length']))
+
 
+def generate_random_valid_username(website, length):
+    for random_username in generate_random_username(website, length):
+        if username_api.check_format(website, random_username):
+            return random_username
+    raise Exception('Cannot generate valid username')
 
-def generate_random_character(length, first_char_letter):
-    for i in range(length):
-        if first_char_letter and i == 0:
-            yield random.choice(string.ascii_letters)
-        else:
-            yield random.choice(string.ascii_letters + string.digits)
+
+def generate_random_username(website, length):
```

Comment:
this should be above `generate_random_valid_username`, as that is the only caller.

Reply:
sure